src/pushi/base/web_push.py

```
@@ -0,0 +1,425 @@
+
+    Notifications are sent using VAPID authentication and require
+    subscription objects containing endpoint, p256dh key, and auth key.
+
+    :see: https://w3c.github.io/push-api
+    """
+
+    def __init__(self, owner):
+        handler.Handler.__init__(self, owner, name="web_push")
+        self.subs = {}
+
+    def send(self, app_id, event, json_d, invalid={}):
```

Ruff (0.14.7)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)